test/interface_alertmanager_test.py

```diff
@@ -0,0 +1,78 @@
+import json
+import sys
+from unittest.mock import (
+    call,
+    MagicMock,
+    patch,
+)
+import unittest
+from uuid import uuid4
+
+sys.path.append('lib')
+sys.path.append('src')
+from interface_alertmanager import (
+    AlertManagerInterface,
+    NewAlertManagerRelationEvent,
+)
+
+
+class NewAlertManagerRelationEventTest(unittest.TestCase):
+
+    @patch('interface_alertmanager.EventBase', spec_set=True)
+    def test__it_snapshots_the_remote_data(self, mock_event_base_cls):
+        # Setup
+        mock_handle = MagicMock()
+        mock_data = {
+            str(uuid4()): str(uuid4()),
+            str(uuid4()): str(uuid4()),
+            str(uuid4()): str(uuid4()),
+        }
+
+        # Exercise
+        new_relation_event = NewAlertManagerRelationEvent(mock_handle,
```

Comment:
Shouldn't we use Harness and `.emit()` instead of calling the event directly?

Reply:
We could but doing that doesn't really add any value to this test since all we cares about here is whether `NewAlertManagerRelationEvent.snapshot()` returns the correct value or not.